engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -460,7 +460,26 @@ protected void restoreOutput(PrintStream originalStream)
     }
   }
 
-  AppPackage newAppPackageInstance(File f) throws IOException, ZipException
+  AppPackage newAppPackageInstance(URI uri, boolean suppressOutput) throws IOException
+  {
+    PrintStream outputStream = suppressOutput ? suppressOutput() : null;
+    try {
+      final String scheme = uri.getScheme();
+      if (scheme == null || scheme.equals("file")) {
+        return new AppPackage(new FileInputStream(new File(expandFileName(uri.getPath(), true))), true);
+      } else {
```

Comment:
It may be safer to delegate to the FileSystem to deal with schemes it supports and throw an Exception for those that it does not support. In the future, if support for "http" or "ftp" and other schemes are added, I'd recommend to handle them similar to "file", by checking for those schemes explicitly.